openmc/lib/material.py

```diff
@@ -87,6 +87,8 @@ class Material(_FortranObjectWithID):
         ID of the material
     nuclides : list of str
         List of nuclides in the material
+    density : float
```

Comment:
Yes, I have to admit that I actually confused myself when writing a test because of this very issue!

By definition, a property can't have an extra argument. I think we should either 1) add a method that allows specifying units like `get_density(units)`, or 2) add another property that gives the density in atom/b-cm. If we go with 2, it would probably make the most sense to have `density` (which gives atom/b-cm) and `density_gpcc` properties to match the C++ interface for `Material`. Let me know which you prefer.

Reply:
Hrmm I'd overlooked that issue with these being properties. Good point.

My vote is for option 1. It is more verbose, but it adds clarity in both the argument and docstrings when users apply `help` to the method.

Any harm in having option 2 in addition to 1?

Reply:
No true "harm" in having both, but I do feel like it violates the "one obvious way to do it" Zen of Python rule so I'd rather have just one way. If we stick with option 1, do you propose getting rid of the `density` property?

Reply:
Fair enough re: Zen of Python. And yeah I suppose along those same lines we'd want to remove the `density` property.

Reply:
Ok, just updated with a `get_density` method that replaces the `density` property